packages/flutter/lib/src/gestures/long_press.dart

```diff
@@ -58,3 +167,107 @@ class LongPressGestureRecognizer extends PrimaryPointerGestureRecognizer {
   @override
   String get debugDescription => 'long press';
 }
+
+/// Recognizes long presses that can be subsequently dragged around.
+///
+/// Similar to a [LongPressGestureRecognizer] where a press has to be held down
+/// at the same location for a long period of time. However drag events that
+/// occur after the long-press hold threshold has past will not cancel the
+/// gesture. The [onLongPressDragUpdate] callback will be called until an up
+/// event occurs.
+///
+/// See also:
+///
+///  * [LongPressGestureRecognizer], which cancels its gesture if a drag event
+///    occurs at any point during the long-press.
+class LongPressDragGestureRecognizer extends PrimaryPointerGestureRecognizer {
```

Comment:
shouldn't this be a DragGestureRecognizer subclass?

Reply:
The DragGestureRecognizer is unconcerned with the time dimension which is importance here. Perhaps both PrimaryPointerGestureRecognizer and/or DragGestureRecognizer can be refactored to be mixins, though that would break more APIs

Reply:
Presumably adding the time dimension would be what you would do in the DragGestureRecognizer subclass.

Reply:
We could add a time component to DragGestureRecognizer or a move component to PrimaryPointerGestureRecognizer. I don't feel strongly either way. We can modify DragGestureRecognizer and have this subclass it instead if you prefer. 

Ideally, the time and position components would both come in via mixins. We can also just not change either subclasses and do it here directly but we'd have to repeat that work for some of the other gestures that are coming up such as double tap drag.